noxfile.py

```diff
@@ -30,6 +30,7 @@
 
 BLACK_VERSION = "black==22.3.0"
 ISORT_VERSION = "isort==5.12.0"
+PYTEST_VERSION = "pytest<=7.4.4"
```

Comment:
nit: add a comment to capture the specific reason for adding an upper limit to `pytest` (with a link to the corresponding issue to follow up) . If there was a bad release of pytest, consider excluding the specific bad release.

Reply:
Added! As far as I can tell from recent test failures it's that https://github.com/str0zzapreti/pytest-retry isn't compatible with 8.x, which was released this weekend.